lib/jobba/statuses.rb

```diff
@@ -36,7 +36,7 @@ def delete_all!
     load
 
     # preload prior attempts because loading them is not `multi`-friendly
-    @cache.each(&:prior_attempts)
+    @cache.each { |cache| cache.prior_attempts.each(&:prior_attempts) }
 
```

Comment:
not sure if this is a actually array but you might have been able to use flat_map .